apps/ledger-live-mobile/src/families/cosmos/operationDetails.tsx

```diff
@@ -192,21 +199,30 @@ function OperationDetailsExtra({ extra, type, account }: Props) {
 
     case "REWARD": {
       const { validators } = extra;
-      if (!validators || validators.length <= 0) break;
-      const validator = extra.validators[0];
-      const formattedValidator = cosmosValidators.find(
-        v => v.validatorAddress === validator.address,
-      );
+      if (validators == null || validators.length <= 0) break;
```

Comment:
Seems like we do this check on every case. You can just put this condition above and return null if true

Reply:
There is also a condition on sourceValidator on Redelegate

Reply:
Also, there is a ticket for each case, see cosmos epic in jira https://ledgerhq.atlassian.net/browse/LIVE-5701

Reply:
Indeed seems like unnecessary duplication to me.

What could be done is something like that, in order not to repeat his check for every case:

```tsx
  const OperationDetailsSection = (
    <>
      {extra.memo && (
        <Section title={t("operationDetails.extra.memo")} value={extra.memo} />
      )}
    </>
  );

  if (!extra.validators || extra.validators.length <= 0) {
    return <>{OperationDetailsSection}</>;
  }

  switch (type) {
    case "DELEGATE": {
      const validator = extra.validators[0];

...

    case "REDELEGATE": {
      const { sourceValidator } = extra;
      if (!sourceValidator) break;

...

  return (
    <>
      {ret}
      {OperationDetailsSection}
    </>
  );
```

But no strong opinion on that

Reply:
Done